File: ccl_v0_1/tests/run_all_tests.py
#!/usr/bin/env python3
from __future__ import annotations

import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
sys.path.insert(0, str(ROOT / "evaluator"))

from reference_evaluator import compare_expected, load_yaml, run_case  # type: ignore


def main() -> None:
    cases_dir = ROOT / "tests" / "cases"
    passed = 0
    failed = 0
    lines = ["# Test Results", ""]
    for case_path in sorted(cases_dir.glob("*.yaml")):
        case = load_yaml(case_path)
        policy_path = ROOT / "policies" / case["policy"]
        result = run_case(policy_path, case_path)
        ok, _detail = compare_expected(result, case["expected"])
        status = "PASS" if ok else "FAIL"
        lines.append(f"- {case_path.name}: **{status}**")
        if ok:
            passed += 1
        else:
            failed += 1
            print(f"{case_path.name}: FAIL")
            print("Result:", result)
            print("Expected:", case["expected"])
    summary = f"Passed: {passed}, Failed: {failed}"
    lines.extend(["", summary, ""])
    print(summary)
    (ROOT / "tests" / "TEST_RESULTS.md").write_text("\n".join(lines), encoding="utf-8")
    if failed:
        raise SystemExit(1)


if __name__ == "__main__":
    main()

File: ccl_v0_1/tests/run_all_tests.js
#!/usr/bin/env node

const fs = require("node:fs");
const path = require("node:path");
const {
  compareExpected,
  loadYaml,
  resolvePolicyPath,
  runCase,
} = require("../evaluator/reference_evaluator.js");

const casesDir = path.resolve(__dirname, "cases");

function main() {
  const caseFiles = fs
    .readdirSync(casesDir)
    .filter((file) => file.endsWith(".yaml"))
    .sort();

  let passed = 0;

  for (const file of caseFiles) {
    const casePath = path.join(casesDir, file);
    const testCase = loadYaml(casePath);
    const policyPath = resolvePolicyPath(casePath, testCase.policy);
    const result = runCase(policyPath, casePath);
    const comparison = compareExpected(result, testCase.expected);

    if (!comparison.ok) {
      console.error(`FAIL ${testCase.name}`);
      console.error(JSON.stringify(comparison.detail, null, 2));
      process.exitCode = 1;
      return;
    }

    passed += 1;
    console.log(`PASS ${testCase.name}`);
  }

  console.log(`\n${passed}/${caseFiles.length} cases passed`);
}

main();
